cmd/ndm_daemonset/controller/blockdevice.go

```diff
@@ -157,6 +157,36 @@ func addBdLabels(bd *apis.BlockDevice, ctrl *Controller) error {
 	return nil
 }
 
+// RelaxedJSONPathExpression attempts to be flexible with JSONPath expressions, it accepts:
+//   * metadata.name (no leading '.' or curly braces '{...}'
+//   * {metadata.name} (no leading '.')
+//   * .metadata.name (no curly braces '{...}')
+//   * {.metadata.name} (complete expression)
+// And transforms them all into a valid jsonpath expression:
+//   {.metadata.name}
+func RelaxedJSONPathExpression(pathExpression string) (string, error) {
```

Comment:
@Ab-hishek Can you put a reference to the original code here.

Reply:
Sure Akhil.